src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -57,18 +57,24 @@ inline void projectOntoSpaceFillingCurve(ExecutionSpace const &space,
       });
 }
 
-template <typename ExecutionSpace, typename Values, typename Nodes>
-inline void initializeSingleLeafNode(ExecutionSpace const &space,
-                                     Values const &values,
-                                     Nodes const &leaf_nodes)
+template <typename ExecutionSpace, typename Values, typename IndexableGetter,
+          typename Nodes, typename BoundingVolume>
+inline void
+initializeSingleLeafTree(ExecutionSpace const &space, Values const &values,
+                         IndexableGetter const &indexable_getter,
+                         Nodes const &leaf_nodes, BoundingVolume &bounds)
 {
   ARBORX_ASSERT(leaf_nodes.extent(0) == 1);
   ARBORX_ASSERT(values.size() == 1);
 
-  Kokkos::parallel_for(
-      "ArborX::TreeConstruction::initialize_single_leaf",
+  Kokkos::parallel_reduce(
+      "ArborX::TreeConstruction::initialize_single_leaf_tree",
       Kokkos::RangePolicy<ExecutionSpace>(space, 0, 1),
-      KOKKOS_LAMBDA(int) { leaf_nodes(0) = makeLeafNode(values(0)); });
+      KOKKOS_LAMBDA(int, BoundingVolume &update) {
+        leaf_nodes(0) = makeLeafNode(values(0));
+        expand(update, indexable_getter(leaf_nodes(0).value));
+      },
+      bounds);
```

Comment:
I thought switching to `parallel_reduce` was a great idea. But it now requires using `Kokkos::Sum<BoundingVolume>` and introducing reduction identity for kDOP.